resources/qml/ColorDialog.qml

```diff
@@ -17,61 +17,133 @@ UM.Dialog
 
     property variant catalog: UM.I18nCatalog { name: "cura" }
 
-    minimumHeight: UM.Theme.getSize("small_popup_dialog").height
-    minimumWidth: UM.Theme.getSize("small_popup_dialog").width / 1.5
-    height: minimumHeight
-    width: minimumWidth
+    margin: UM.Theme.getSize("default_margin").width
+
+    property alias swatchGridColumns: colorSwatchGrid.columns
+
+    // In this case we would like to let the content of the dialog determine the size of the dialog
+    // however with the current implementation of the dialog this is not possible, so instead we calculate
+    // the size of the dialog ourselves.
+    minimumWidth: content.width + 2 * margin
+    minimumHeight:
+        content.height                                 // content height
+      + buttonRow.height                               // button row height
+      + 3 * margin                                     // top and bottom margin and margin between buttons and content
 
     property alias color: colorInput.text
+    property var swatchColors: [
+        "#2161AF", "#57AFB2", "#F7B32D", "#E33D4A", "#C088AD",
+        "#5D88BE", "#5ABD0E", "#E17239", "#F74E46", "#874AF9",
+        "#50C2EC", "#8DC15A", "#C3977A", "#CD7776", "#9086BA",
+        "#FFFFFF", "#D3D3D3", "#9E9E9E", "#5A5A5A", "#000000",
+    ]
 
-    margin: UM.Theme.getSize("default_margin").width
-    buttonSpacing: UM.Theme.getSize("default_margin").width
+    Component.onCompleted: updateSwatches()
+    onSwatchColorsChanged: updateSwatches()
 
-    UM.Label
+    function updateSwatches()
     {
-        id: colorLabel
-        font: UM.Theme.getFont("large")
-        text: catalog.i18nc("@label", "Color Code (HEX)")
+        swatchColorsModel.clear();
+        for (const swatchColor of base.swatchColors)
+        {
+            swatchColorsModel.append({ swatchColor });
+        }
     }
 
-    TextField
+    Column
     {
-        id: colorInput
-        text: "#FFFFFF"
-        selectByMouse: true
-        anchors.top: colorLabel.bottom
-        anchors.topMargin: UM.Theme.getSize("default_margin").height
-        onTextChanged: {
-            if (!text.startsWith("#"))
+        id: content
+        width: childrenRect.width
+        height: childrenRect.height
+        spacing: UM.Theme.getSize("wide_margin").height
+
+        GridLayout {
+            id: colorSwatchGrid
+            columns: 5
+            width: childrenRect.width
+            height: childrenRect.height
+            columnSpacing: UM.Theme.getSize("thick_margin").width
+            rowSpacing: UM.Theme.getSize("thick_margin").height
+
+            Repeater
             {
-                text = `#${text}`;
+                model: ListModel
+                {
+                    id: swatchColorsModel
+                }
+
+                delegate: Rectangle
+                {
+                    color: swatchColor
+                    width: 24
```

Comment:
Not really a big issue, but shouldn't these be themed? Right now the sizes will be different depending on the DPI

Reply:
Would you introduce these _new_ sizes? They are the same as a "wide margin" but it doesn't feel right to pick those

Reply:
Maybe button_icon? Or extruder icon?